apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -366,6 +366,13 @@ type EnvoyListenerConfig struct {
 	// +optional
 	ConnectionBalancer string `json:"connectionBalancer,omitempty"`
 
+	// Defines the maximum requests for downstream connections. If not specified, there is no limit.
+	// see https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/protocol.proto#envoy-v3-api-msg-config-core-v3-httpprotocoloptions
+	// for more information.
+	//
+	// +optional
+	MaxRequestsPerConnection *uint32 `json:"maxRequestsPerConnection,omitempty"`
```

Comment:
we might want to make these `int32` instead and add a minimum of 0

see https://github.com/projectcontour/contour/issues/2123

Reply:
not sure setting as min of 0 would work, by default, this is unset on envoy. adding 0 might result in undefined behaviour.

Reply:
ah fair point, this field is not like a timeout where setting to 0 means to disable it, so a minimum of 1 would make sense then to make sure we're not getting an invalid value (negative or 0), the fact the field is a pointer is our signal if it has been set or not

hm, maybe ok to leave as-is actually we do have uints scattered around

Reply:
i guess generally we should have a kubebuilder validation minimum of 1 for this field since 0 doesn't make any sense